src/components/screens/forms/form-replies-screen.tsx

```diff
@@ -0,0 +1,278 @@
+/* eslint-disable react/no-unstable-nested-components */
+import { FormControlLabel, Switch, Typography } from "@mui/material";
+import { DataGrid, GridActionsCellItem, GridColDef } from "@mui/x-data-grid";
+import Api from "api";
+import { useApiClient, useAppSelector } from "app/hooks";
+import { ErrorContext } from "components/contexts/error-handler";
+import ConfirmDialog from "components/generic/confirm-dialog";
+import NavigationTab from "components/layouts/navigations/navigation-tab";
+import { Metaform, MetaformField, MetaformFieldType, Reply } from "generated/client";
+import strings from "localization/strings";
+import moment from "moment";
+import React, { useContext, useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { NavigationTabContainer } from "styled/layouts/navigations";
+import { AdminFormRepliesScreenStack, AdminFormRepliesScreenText } from "styled/react-components/react-components";
+import DeleteIcon from "@mui/icons-material/Delete";
+import { selectKeycloak } from "features/auth-slice";
+import { ApplicationRoles, ReplyStatus } from "types";
+
+/**
+ * Form replies screen component
+ */
+const FormRepliesScreen: React.FC = () => {
+  const errorContext = useContext(ErrorContext);
+  const keycloak = useAppSelector(selectKeycloak);
+  
+  const apiClient = useApiClient(Api.getApiClient);
+  const { repliesApi, metaformsApi } = apiClient;
+
+  const [ rows, setRows ] = useState<any[]>([]);
+  const [ loading, setLoading ] = useState(false);
+  const [ managementListColumns, setManagementListColumns ] = useState<MetaformField[]>();
+  const [ metaform, setMetaform ] = useState<Metaform>();
+  const [ deletableReplyId, setDeletableReplyId ] = useState<string | undefined>(undefined);
+  const [ showAllReplies, setShowAllReplies ] = useState(false);
+
+  const useparams = useParams();
+  const { formId } = useparams;
+
+  /**
+   * Builds a row for the table
+   * 
+   * @param reply reply 
+   */
+  const buildRow = (reply: Reply, fields: MetaformField[]) => {
+    const row : { [key: string]: string | number } = {};
+    
+    row.replyId = reply.id!;
+
+    fields.forEach(field => {
```

Comment:
Couldn't this be replaced by map?

Reply:
It doesn't return anything so map would trigger sonar cloud error